### ISSUE TITLE
Type mismatch error

### ISSUE THREAD

**Issue by mayinrain** [abi/screenshot-to-code#384]
**Describe the bug**
"When I upload a video with a .mov extension, the file is recognized as an image format.
**To Reproduce**
1. Go to http://localhost:5173/
2. Click on file uploader
3. Select a file with a .mov extension
4. The file is recognized as an img.

**Screenshots of backend AND frontend terminal logs**
I printed the Base64 of the file and found that the prefix is data:application/octet-stream, so it is recognized as an img.
 
<img width="525" alt="image" src="https://github.com/user-attachments/assets/b1420361-8cb2-43f1-ae51-9cc610c0e179">
Here is the corresponding code

**Comment by abi** [comment]
Yeah, I would recommend converting .mov to .mp4 before uploading it.

We probably will support more video formats, but there's lots of different ones so that isn't the biggest focus at the moment.

**Comment by Jefsky** [comment]
Submitted a fix in #601. The root cause was that Firefox reports an empty MIME type for .mov files, so the existing MIME type fallback in fileToDataURL was skipped. The fix infers the MIME type from the file extension when file.type is empty.
